test/e2e/service.go

```diff
@@ -2350,3 +2371,33 @@ func execSourceipTest(f *framework.Framework, c *client.Client, ns, nodeName, se
 
 	return execPodIp, sourceIp
 }
+
+func getLoadBalancerName(service *api.Service) string {
+	//GCE requires that the name of a load balancer starts with a lower case letter.
+	ret := "a" + string(service.UID)
+	ret = strings.Replace(ret, "-", "", -1)
+	//AWS requires that the name of a load balancer is shorter than 32 bytes.
+	if len(ret) > 32 {
+		ret = ret[:32]
+	}
+	return ret
+}
+
+func cleanupServiceGCEResources(loadBalancerName string) {
+	if pollErr := wait.Poll(5*time.Second, lbCleanupTimeout, func() (bool, error) {
+		if err := framework.CleanupGCEResources(loadBalancerName); err != nil {
```

Comment:
we're adding a test that deletes an lb service, that's why i tagged Girish: https://github.com/kubernetes/kubernetes/pull/31991/files#diff-20a4e2095b63ecd60dd25e78bcd67372R1232